support-workers/src/main/scala/com/gu/zuora/subscriptionBuilders/DigitalSubscriptionCorporateRedemptionBuilder.scala

```diff
@@ -0,0 +1,60 @@
+package com.gu.zuora.subscriptionBuilders
+
+import java.util.UUID
+
+import cats.data.EitherT
+import cats.implicits._
+import com.gu.support.config.TouchPointEnvironment
+import com.gu.support.redemption.corporate.CorporateCodeValidator
+import com.gu.support.redemption.{InvalidCode, InvalidReaderType, ValidCorporateCode}
+import com.gu.support.redemptions.RedemptionData
+import com.gu.support.workers.DigitalPack
+import com.gu.support.workers.ProductTypeRatePlans.digitalRatePlan
+import com.gu.support.zuora.api.ReaderType.Corporate
+import com.gu.support.zuora.api.{ReaderType, SubscriptionData}
+import com.gu.zuora.subscriptionBuilders.ProductSubscriptionBuilders.{buildProductSubscription, validateRatePlan}
+import org.joda.time.LocalDate
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DigitalSubscriptionCorporateRedemptionBuilder(
+  codeValidator: CorporateCodeValidator,
+  today: () => LocalDate,
+) {
+
+  def build(
+    redemptionData: RedemptionData,
+    digitalPack: DigitalPack,
+    requestId: UUID,
+    environment: TouchPointEnvironment,
+  )(implicit ec: ExecutionContext): EitherT[Future, InvalidCode, SubscriptionData] = {
```

Comment:
this and the below are split out from the old DigitalSubscriptionBuilder object.
The main benefit is they don't need to conform to a common type so we have got rid of the old
`type BuildResult = EitherT[Future, Either[PromoError, InvalidCode], SubscriptionData]`

Reply:
🎉  that was a Frankenstein type!

Reply:
a Frankentype